infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutAssetDao.kt

```diff
@@ -341,6 +347,34 @@ abstract class LayoutAssetDao<T : LayoutAsset<T>>(
             }
     }
 
+    override fun updateImplementedDesignDraftReferences(
```

Comment:
Tästä kommentteja? Ratkaisu on nyt siis se että design-draft olio laitetaan osoittamaan oikeaan riviin. Toinen vaihtoehto olisi vain poistaa koko design-draft. Molemmissa olisi varmasti puolensa, enkä ole ihan varma kumpi on oikeampi. Pitäisikö design-draft poistua jos official-designista viety muutos julkaistaan main-puolella?

Reply:
Kokeilin, saisiko tämän koko rumban vältettyä jonkinlaisella muutoksella in_layout_context-funktioihin ja mahdollisesti indekseihin, ja ehkä teoriassa saisi, mutta ei hyvällä perffillä ja nätisti.

Tämä ratkaisu taitaa siis olla yhä se, joka on eniten oikein, vaikka onkin yhdestä näkökulmasta vähän outo, nimittäin: Nyt mainin julkaisun yhteydessä yhtäältä live-taulusta poistetaan se merkityksellisempi design-official-rivi, mutta toisaalta saatetaan jättää olemaan vähemmän merkityksellinen hiekkalaatikkopuolen design-draft-rivi. Mutta tällä ratkaisulla saadaan estettyä tilanne, jossa julkaisu mainissa voisi tuhota käyttäjän tekemiä muutoksia, ja tiedon ehjyys on tärkeämpää kuin tuon yhden sisäisen outouden välttäminen.

Reply:
Vaikea keissi kyllä. Tuntuisi mielestäni todella oudolta että jokin (operaattorin näkökulmasta) puhtaasti main-branchissa tehty muutos tuhoaisi asioita designista, joten sikäli tämä ratkaisu lienee paras. Sikäli tämäkin on tosin hieman pelottava, että nyt meillä on käytännössä funktio jota sopivasti kutsumatta jättämällä saadaan kanta aika ikävään tilaan. Mietin olisikohan tämän lopputilaa mahdollista enforcata jotenkin ulkoapäin esim. kantarajoituksella?

Reply:
Kirjataan tännekin jotta jää jälki: kantarajoitus olisi ehdottomasti kiva mutta sellainen on hankala saada aikaa, koska tuossa joutuu tarkastelemaan viitatun rivin muuta tilaa nähdäkseen että onko se design/main. Eli paremman puutteessa, tällä mennään.